js/components/skipControls.js

```diff
@@ -98,25 +98,27 @@ class SkipControls extends React.Component {
   }
 
   /**
-   * Determines whether or not the current video is at the live edge based on the
-   * playhead state and duration.
+   * Determines whether or not the current video is at the end (VOD) or at the
+   * live edge (DVR Live Streams) based on the playhead state and duration.
    * @private
-   * @return {Boolean} True if the video is at the live edge, false otherwise.
-   * Note: This function always returns false for VOD.
+   * @return {Boolean} True if the video is at the video end/live edge, false otherwise.
    */
-  isAtLiveEdge() {
+  isAtVideoEnd() {
     const isLiveStream = Utils.getPropertyValue(
       this.props.controller,
       'state.isLiveStream',
       false
     );
+    let isVideoEnd = false;
+    const duration = Utils.getPropertyValue(this.props.controller, 'state.duration', 0);
+    const currentPlayhead = Utils.ensureNumber(this.props.currentPlayhead, 0);
+
     if (isLiveStream) {
-      const duration = Utils.getPropertyValue(this.props.controller, 'state.duration', 0);
-      const currentPlayhead = Utils.ensureNumber(this.props.currentPlayhead, 0);
-      const isLiveNow = Math.abs(currentPlayhead - duration) < 1;
-      return isLiveNow;
+      isVideoEnd = Math.abs(currentPlayhead - duration) < 1;
+    } else {
+      isVideoEnd = currentPlayhead >= duration;
     }
-    return false;
+    return isVideoEnd;
```

Comment:
Let me suggest a bit shorted and more self explanatory code that doing the same:
```suggestion
      const duration = Utils.getPropertyValue(this.props.controller, 'state.duration', 0);
      const currentPlayhead = Utils.ensureNumber(this.props.currentPlayhead, 0);
      const isAtLiveEdge = isLiveStream && Math.abs(currentPlayhead - duration) < 1;
      const isAtVideoEnd = !isLiveStream && currentPlayhead >= duration;
      return isAtLiveEdge || isAtVideoEnd;
```

Reply:
Also I'm not sure if I got right the root case, but I saw 4 seconds gap before video end, that should be considered (from your comment in the ticket), and I don't see you consider this gap in the code. Aren't we missing something like `currentPlayhead >= duration - 4;`?

Reply:
@zelibobla I disagree with that being a more self explanatory code. The purpose of the chosen syntax is to improve readability, not to minimize the lines of code.

Reply:
@zelibobla as mentioned on the PR description, this update is not addressing the root cause, it's just covering the UI side of the problem. The root cause will be addressed on a separate PR for video-plugins-internal